File: Linguagem de Programação I/Extras/Beecrowd/2674.py
def is_prime(n):
    if n < 2:
        return False
    for i in range(2, int(n**0.5) + 1):
        if n % i == 0:
            return False
    return True

def is_super(n):
    prime_digits = {'2', '3', '5', '7'}
    return all(digit in prime_digits for digit in str(n))

while True:
    try:
        n = int(input())
        if is_prime(n):
            if is_super(n):
                print('Super')
            else:
                print('Primo')
        else:
            print('Nada')
    except EOFError:
        break

File: Linguagem de Programação I/Extras/Beecrowd_Extra/2674.cpp
#include <iostream>
#include <cmath>
#include <string>

bool is_prime(int n) {
    if (n < 2) {
        return false;
    }
    for (int i = 2; i <= sqrt(n); i++) {
        if (n % i == 0) {
            return false;
        }
    }
    return true;
}

bool is_super(int n) {
    std::string prime_digits = "2357";
    std::string str_n = std::to_string(n);

    for (char c : str_n) {
        if (prime_digits.find(c) == std::string::npos) {
            return false;
        }
    }
    return true;
}

int main() {
    int n;
    while (std::cin >> n) {
        if (is_prime(n)) {
            if (is_super(n)) {
                std::cout << "Super\n";
            } else {
                std::cout << "Primo\n";
            }
        } else {
            std::cout << "Nada\n";
        }
    }
    return 0;
}